src/main/scala/model/Kinds.scala

```diff
@@ -0,0 +1,113 @@
+package model
+
+import model.Alleles.AlleleKind
+import model.Genes.GeneKind
+
+import scala.language.implicitConversions
+import scala.util.Random
+
+/**
+ * An Enumeration for all the Alleles present in the World.
+ */
+object Alleles extends Enumeration{
+  type AlleleKind = Value
+
+  /**
+   * The information each AlleleKind must have
+   * @param dominant specifies if it's dominant or not, it could be empty it not chosen yet
+   * @param locked   show if the dominance has already been chosen (becoming unmodifiable) or not
+   */
+  protected case class AllelesVal(private var dominant: Option[Boolean] = Option.empty,
+                                  var locked: Boolean = false) extends super.Val {
+    def setDominance(cond: Boolean): Unit = {
+      if (locked) throw new MultipleDominanceAssignmentException else {
+        dominant = Option(cond)
+        locked = true
+      }
+    }
+    def isDominant: Option[Boolean] = dominant
+  }
+  implicit def valueToAllelesVal(x: Value): AllelesVal = x.asInstanceOf[AllelesVal]
+
+  val WHITE_FUR: AllelesVal = AllelesVal()
+  val BROWN_FUR: AllelesVal = AllelesVal()
+  val LONG_FUR: AllelesVal = AllelesVal()
+  val SHORT_FUR: AllelesVal = AllelesVal()
+  val LONG_TEETH: AllelesVal = AllelesVal()
+  val SHORT_TEETH: AllelesVal = AllelesVal()
+  val HIGH_EARS: AllelesVal = AllelesVal()
+  val LOW_EARS: AllelesVal = AllelesVal()
+  val HIGH_JUMP: AllelesVal = AllelesVal()
+  val LOW_JUMP: AllelesVal = AllelesVal()
```

Comment:
per queste costanti invece è più corretto trasferirle nel file a parte o lasciarle qui essendo calcolate attraverso il metodo _AllelesVal_ ? Io personalmente le trasferirei nel file dedicato ma vorrei avere anche la vostra opinione @aslucc @anitvam @martaSpadoni

Reply:
Secondo me è giusto lasciarle qui perchè queste non sono costanti del sistema ma costanti dei tipi. Esempi di costanti del sistema: Numero di generazioni massimo, Generazioni di vita dei conigli, Numero di conigli massimo, Numero di figli per coppia di coniglio ecc.

Reply:
Perfetto ora mi è più chiara la divisione che vogliamo fare tra costanti del sistema e non!

Reply:
Io ho anche pensato che Allele e Genes sono un punto in cui sarebbe bello offrire estendibilità, che sia possibile per un eventuale sviluppatore modificarli mettendo dei suoi Alleli e Geni custom perciò non possono essere costanti di sistema